File: LeetCode/LeetCode Practice Problems/1460- Make Two Arrays Equal by Reversing Sub-arrays.cpp
class Solution {
public:
    bool canBeEqual(vector<int>& target, vector<int>& arr) {
        map<int,int> m;
        
        for(auto i: target) m[i]++;
        
        for(auto i: arr) m[i]--;
        
        for(auto i: m)
            if(i.second != 0)
                return false;
        
        return true;
    }
};

File: LeetCode/LeetCode Practice Problems/1460-Make Two Arrays Equal by Reversing Sub-arrays.java
class Solution {
    public boolean canBeEqual(int[] target, int[] arr) {
        Arrays.sort(arr);
        Arrays.sort(target);
        
        for(int i=0;i<arr.length;i++) {
            if(arr[i] != target[i]) {
                return false;
            }
        }
        
        return true;
    }
}
